src/lsdb/io/to_hipscat.py

```diff
@@ -119,11 +119,25 @@ def write_partitions(
     pixel_to_result_index = {}
 
     partitions = catalog._ddf.to_delayed()
+
+    # Get the indices of non-empty partitions
+    partition_sizes = catalog._ddf.map_partitions(len).compute()
```

Comment:
So this works, but it needs the entire catalog to be computed twice (once to get the length of each partition, and one to actually write the results. Could we instead move this length check inside the `perform_write` method?

Then when we compute the partition_sizes, we can perform the `len(non_empty_partitions)` check there.